### PR TITLE
refactor: Rename UserName to QualName

### PR DESCRIPTION
**Pull request by croyzor**

### CODE REVIEW

File: brat/Brat/Checker/Helpers.hs [hunk 1]
```diff
@@ -131,7 +131,7 @@ pullPorts toPort showFn (p:ports) types = do
      else pure (x, xs)
    | otherwise = second (x:) <$> pull1Port p xs
 
-combineDisjointEnvs :: M.Map UserName v -> M.Map UserName v -> Checking (M.Map UserName v)
+combineDisjointEnvs :: M.Map QualName v -> M.Map QualName v -> Checking (M.Map QualName v)
```
Comment (acl-cqc) [inline brat/Brat/Checker/Helpers.hs:134]: driveby: we should really use `throwLeft` for this

File: brat/Brat/Checker/Monad.hs [hunk 2]
```diff
@@ -139,29 +139,29 @@ throwLeft :: Either ErrorMsg a -> Checking a
 throwLeft (Right x) = pure x
 throwLeft (Left msg) = err msg
 
-vlup :: UserName -> Checking [(Src, BinderType Brat)]
+vlup :: QualName -> Checking [(Src, BinderType Brat)]
```
Comment (acl-cqc) [inline brat/Brat/Checker/Monad.hs:142]: Oooh, that's odd. We can have qualified names of variables?

File: brat/Brat/Checker/Monad.hs [hunk 3]
```diff
@@ -172,7 +172,7 @@ tlup (m, c) = req (TLup (m, c)) >>= \case
     Brat -> Kernel
     Kernel -> Brat
 
-lookupAndUse :: UserName -> KEnv
+lookupAndUse :: QualName -> KEnv
```
Comment (acl-cqc) [inline brat/Brat/Checker/Monad.hs:175]: Similarly, if it's something that might be used then will it really be a QualName?

**Review by acl-cqc — APPROVED** [review]
A couple of comments, but some are definitely for another PR.

I'm not unhappy with QualName, but other possibilities - QualId(entifier), UserId, UserVarName (too long!)

File: brat/Brat/Checker/Monad.hs (hunk 3 above)
Comment (croyzor) [reply brat/Brat/Checker/Monad.hs:175]: variables are parsed as qualified names, so they can include prefixes

File: brat/Brat/Checker/Monad.hs (hunk 2 above)
Comment (acl-cqc) [reply brat/Brat/Checker/Monad.hs:142]: Ah right yes thank you. Ok - so `vlup` is really the misnamed one - it does not necessarily look up variables; it just looks up names. Really should be `nlup` or `qnlup` or just `lup` (or `blup` to distinguish from `klup`? Hey, that one's actually pronouncable in one syllable... 😁 😀 )

File: brat/Brat/Checker/Monad.hs (hunk 3 above)
Comment (acl-cqc) [reply brat/Brat/Checker/Monad.hs:175]: Even kernel names? Can declare a variable of type qubit whose name has a prefix?

Comment (croyzor) [reply brat/Brat/Checker/Monad.hs:175]: Ah I see, we could be more strict about kernel variables, you're right
